### PR TITLE
python310Packages.nianet: init at 1.1.4

### PR DESCRIPTION
**Pull request by firefly-cpp**
###### Description of changes

Python package for designing and constructing neural network topologies using nature-inspired algorithms: [LINK](https://github.com/SasoPavlic/NiaNet) 

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by firefly-cpp** [comment]
@SuperSandro2000, I would like to ask you for one review if you have time. Many thanks for considering my request.

**Comment by firefly-cpp** [comment]
@fabaff, please review if possible. Thanks.

File: pkgs/development/python-modules/nianet/default.nix [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  propagatedBuildInputs = [
+    niapy
+    numpy
+    scikit-learn
+    torch
+  ];
+
+  # no serious tests | some tests are incorrect | skip for now
+  # reported to the upstream
+  #checkInputs = [
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/nianet/default.nix:41]: Those seem to be fine but small https://github.com/SasoPavlic/NiaNet/blob/main/tests/test_nianet.py

Comment (firefly-cpp) [reply pkgs/development/python-modules/nianet/default.nix]: Nianet upgraded to the latest release. Tests are now enabled.

Comment (firefly-cpp) [reply pkgs/development/python-modules/nianet/default.nix]: @SuperSandro2000, thanks.

File: pkgs/development/python-modules/nianet/default.nix [hunk 2]
```diff
@@ -8,24 +8,26 @@
...
   format = "pyproject";
 
   disabled = pythonOlder "3.6";
 
   src = fetchFromGitHub {
     owner = "SasoPavlic";
     repo = pname;
-    rev = "Latest";
-    sha256 = "sha256-df3kOWs17fT74dyKF2ZqPNOhMeH8yDEDkJOsMtFNHsM=";
+    rev = "version_1.1.4";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/nianet/default.nix:25]:
```suggestion
    rev = "version_${version}";
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
@ofborg build python310Packages.nianet

**Pull request merged**